.github/workflows/release.yml

```diff
@@ -22,11 +22,12 @@ jobs:
         with:
           # Allow goreleaser to access older tag information.
           fetch-depth: 0
+          persist-credentials: false
       - uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
         with:
           go-version-file: 'go.mod'
-          cache: true
-      - uses: grafana/shared-workflows/actions/get-vault-secrets@main
+          cache: false
+      - uses: grafana/shared-workflows/actions/get-vault-secrets@75804962c1ba608148988c1e2dc35fbb0ee21746 # main as of 28/04/2025
```

Comment:
You could also pin this to the sha relating to the latest `get-vault-secrets` tag. Latest tag for that action is `get-vault-secrets-v1.1.0` (sha is `28361cdb22223e5f1e34358c86c20908e7248760`). We should be able to set up some automation on those, but I don't think dependabot supports that (renovate might).

Reply:
unsure tbh if we should be also pinning against _our_ (grafana) workflows (of course not pinning it assumes the risk of attacks against our own tagging).

See for example today's PR for grafana (under the context of the current incident) https://github.com/grafana/grafana/pull/104657/files#diff-ef11c2eba36ab12f8b9d4f842ddebabe52ac5e781a112f0f68776c0b911e8c04
```diff
- uses: grafana/shared-workflows/actions/get-vault-secrets@get-vault-secrets-v1.0.1
+ uses: grafana/shared-workflows/actions/get-vault-secrets@get-vault-secrets-v1.0.1 # zizmor: ignore[unpinned-uses]
```

Reply:
So suggest pinning to the `tag`, accepting the risk of the tag being poisoned, and tell `zizmor` to ignore it?  I'm ok with that.